src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -161,13 +160,19 @@ end
 Create an undirected [cycle graph](https://en.wikipedia.org/wiki/Cycle_graph)
 with `n` vertices.
 """
-function CycleGraph(n::Integer)
-    g = SimpleGraph(n)
-    for i = 1:(n - 1)
-        add_edge!(g, SimpleEdge(i, i + 1))
+function CycleGraph(n::T) where {T <: Integer}
+    n <= 1 && return SimpleGraph(n)
+    n == 2 && return SimpleGraph(Edge{T}.([(1, 2)]))
+
+    ne = Int(n)
+    fadjlist = Vector{Vector{T}}(undef, n)
+    @inbounds fadjlist[1] = T[2, n]
+    @inbounds fadjlist[n] = T[1, n-1]
+
+    @inbounds @simd for u = 2:(n-1)
+        fadjlist[u] = T[u-1, u+1]
```

Comment:
I don't like this. It relies on `fadjlist` which is an internal data structure, not meant for use except in extreme circumstances. Why do we need to change this function / why can't we use the new "add edges from list"?